services/py-api/src/database/repository/teams_repository.py

```diff
@@ -96,3 +96,15 @@ async def get_verified_registered_teams_count(self) -> int:
         # Ignoring mypy type due to mypy err: 'Returning Any from function declared to return "int"  [no-any-return]'
         # which is not true
         return await self._collection.count_documents({"is_verified": True})  # type: ignore
+    
+    async def fetch_team_by_name(self, team_name: str) -> Result:
+        """Fetches a team by the team_name from the participant"""
+        try:
+            team = await self._collection.find_one({"name": team_name})
+            
+            if team is None:  # If no team is found, return an Err
+                return Err(Exception(f"Team with name '{team_name}' not found"))
```

Comment:
There is a custom exception for this purpose. You can find it in exceptions.py. 
So instead of this you can return Err(TeamNotFoundError()).

Reply:
Fixed in commit a9a6b60a1c533cba63bd89fc5bee4362218c00f7